Lib/test/test_call.py

```diff
@@ -945,5 +945,48 @@ def test_function_with_many_args(self):
                 self.assertEqual(l['f'](*range(N)), N//2)
 
 
+class TestPEP590(unittest.TestCase):
+    def test_sys_not_callable(self):
+        import sys
+
+        self.assertFalse(callable(sys))
+        self.assertFalse(hasattr(sys, "__call__"))
+        self.assertRaisesRegex(TypeError, "Module 'sys' is not callable", sys)
+
+    def test_sys_add_callable(self):
+        try:
+            def exponent(value=42, *, power=1):
+                return value ** power
+
+            sys.__call__ = exponent
+
+            self.assertTrue(callable(sys))
+            self.assertTrue(hasattr(sys, "__call__"))
+
+            self.assertEqual(sys(), 42)
+            self.assertEqual(sys(3), 3)
+            self.assertEqual(sys(power=2), 1764)
+
+        finally:
+            if hasattr(sys, "__call__"):
+                del sys.__call__
+
+    def test_callable_modules(self):
+        from . import callable_module_a, callable_module_b, callable_module_c
```

Comment:
I'd prefer inlining these instead of polluting the `Lib/test` namespace with the added files. Alternatively, put them in a subdirectory.